### PR TITLE
[MRG] ENH updated gitwash

### PR DESCRIPTION
**Pull request by NelleV**
[gitwash](https://github.com/matthew-brett/gitwash)  is a project that maintains documentation on different git workflows.
We have been editing files that are automatically generated instead of updating the files. I have expressed in the past some concerns about having git documentation in the middle of our own instead of pointing to external documentation. I not only think that we are less likely to create good git documentation, but more importantly, it is documentation we need to maintain and in practice isn't really maintained.
I know my point of view is not shared by some developers so I expect some debates on this PR.

I've regenerate the gitwash documentation by downlading `gitwash_dumper.py` typing 

``` shell
python gitwash_dumper.py . matplotlib --project-url=http://matplotlib.org \
            --repo-name matplotlib --github-user matplotlib \
            --project-ml-url=https://mail.python.org/mailman/listinfo/matplotlib-users
```

Most of the changes are quite minor, but we are loosing some "matplotlib" specific examples, which are replaced by generic examples.

**Todo list**
- [x] Document the fact gitwash is generated and should not be modified.

### CODE REVIEW

**Comment by tacaswell** [comment]
This removes all of the re-base documentation that just got merged ;)

**Comment by NelleV** [comment]
@tacaswell I know it does :(
But there is new rebase documentation!

**Comment by NelleV** [comment]
Or maybe there isn't…

**Comment by NelleV** [comment]
After discussion with @efiring and @dopplershift we think the way to move forward is port some of the changes made here into git-wash (such as the rebase part), move out anything that is not gitwash specific outside of the gitwash folder and to start following gitwash again.
That's going to take a bit of work but should lower the maintenance burden of this section.

**Comment by NelleV** [comment]
gitwash already has a rebase tutorial, so no need to port this back…

File: doc/devel/README [hunk 1]
```diff
@@ -0,0 +1,5 @@
+The gitwash folder is auto-generated from
```
Comment (QuLogic) [inline doc/devel/README]: Generated by what?

**Comment by QuLogic** [comment]
Please update when matthew-brett/gitwash#9 is merged.

**Comment by NelleV** [comment]
Maybe we can merge this and update the gitwash when the new release is out?
This PR has been open a while.

**Comment by NelleV** [comment]
This pull request has been opened too long. I am closing it.

**Comment by QuLogic** [comment]
The gitwash PR was merged if you want to re-open this one.

**Pull request closed without merging**